Terminal.Gui/View/Layout/ViewLayout.cs

```diff
@@ -7,55 +7,85 @@
+	/// values.
 	/// </summary>
 	Computed
 }
 
 public partial class View {
-	// The frame for the object. Relative to the SuperView's Bounds.
+	bool _autoSize;
 	Rect _frame;
+	Dim _height = Dim.Sized (0);
+	Dim _width = Dim.Sized (0);
+	Pos _x = Pos.At (0);
+	Pos _y = Pos.At (0);
 
 	/// <summary>
-	/// Gets or sets location and size of the view. The frame is relative to the <see cref="SuperView"/>'s <see cref="Bounds"/>.
+	/// Gets or sets the absolute location and dimension of the view.
 	/// </summary>
-	/// <value>The rectangle describing the location and size of the view, in coordinates relative to the <see cref="SuperView"/>.</value>
+	/// <value>
+	/// The rectangle describing absolute location and dimension of the view,
+	/// in coordinates relative to the <see cref="SuperView"/>'s <see cref="Bounds"/>.
+	/// </value>
 	/// <remarks>
-	/// <para>
-	/// Change the Frame when using the <see cref="LayoutStyle.Absolute"/> layout style to move or resize views.
-	/// </para>
-	/// <para>
-	/// Altering the Frame will change <see cref="LayoutStyle"/> to <see cref="LayoutStyle.Absolute"/>.
-	/// Additionally, <see cref="X"/>, <see cref="Y"/>, <see cref="Width"/>, and <see cref="Height"/> will be set
-	/// to the values of the Frame (using <see cref="Pos.PosAbsolute"/> and <see cref="Dim.DimAbsolute"/>).
-	/// </para>
-	/// <para>
-	/// Altering the Frame will eventually (when the view is next drawn) cause the <see cref="LayoutSubview(View, Rect)"/>
-	/// and <see cref="OnDrawContent(Rect)"/> methods to be called.
-	/// </para>
+	///         <para>
+	///         Frame is relative to the <see cref="SuperView"/>'s <see cref="Bounds"/>.
+	///         </para>
+	///         <para>
+	///         Setting Frame will set <see cref="X"/>, <see cref="Y"/>, <see cref="Width"/>, and <see cref="Height"/>
+	///         to the values of the corresponding properties of the <paramref name="value"/> parameter.
+	///         </para>
+	///         <para>
+	///         This causes <see cref="LayoutStyle"/> to be <see cref="LayoutStyle.Absolute"/>.
+	///         </para>
+	///         <para>
+	///         Altering the Frame will eventually (when the view hierarchy is next laid out via  see cref="LayoutSubviews"/>)
+	///         cause <see cref="LayoutSubview(View, Rect)"/> and <see cref="OnDrawContent(Rect)"/> methods to be called.
+	///         </para>
 	/// </remarks>
-	public virtual Rect Frame {
+	public Rect Frame {
 		get => _frame;
 		set {
 			_frame = new Rect (value.X, value.Y, Math.Max (value.Width, 0), Math.Max (value.Height, 0));
-			//X = _frame.X;
-			//Y = _frame.Y;
-			//Width = _frame.Width;
-			//Height = _frame.Height;
-			if (IsInitialized || LayoutStyle == LayoutStyle.Absolute) {
+
+			// If Frame gets set, by definition, the View is now LayoutStyle.Absolute, so
+			// set all Pos/Dim to Absolute values.
+			_x = _frame.X;
+			_y = _frame.Y;
+			_width = _frame.Width;
+			_height = _frame.Height;
```

Comment:
This is so bad. `ColorPicker` scenario is one of that prove this code isn't correct. If the intention was move this code from `OnAdded` to here was a very wrong choice, maybe on the `BeginInit` or `EndInit`, but never on the `Frame` property. If some tests was changed to reflect this is very wrong. The `backgroundColorPicker.X` is a `Pos.Combine` and ends up loosing his location. I really don't know why the needed to convert a `Pos/Dim`, not `Absolute`, to an `Absolute` value. The Frame must be set after `SetRelativeLayout` finish the calculation with absolute values, but the `Pos/Dim` must always preserve his value when they aren't `Absolute`.

Reply:
Ok sorry, I' seeing that you only set the `_frame` and don't call the `Frame` property anymore in the `SetRelativeLayout` method where you only set the `Pos/Dim` if they are `Absolute`. So, setting directly the `Frame` overwrites the `Pos/Dim` and that begin making sense to me. Good work. Probably many views and scenarios needed/need to be changed to reflect this new behavior.

Reply:
> Ok sorry, I' seeing that you only set the `_frame` and don't call the `Frame` property anymore in the `SetRelativeLayout` method where you only set the `Pos/Dim` if they are `Absolute`. So, setting directly the `Frame` overwrites the `Pos/Dim` and that begin making sense to me. Good work. Probably many views and scenarios needed/need to be changed to reflect this new behavior.

I think I've fixed all those issues. Or at least the majority of them. 

Please keep testing. 

Also, please `./docfx/build.ps1` and click on  http://localhost:8080.

Read and review all related API docs. I worked hard to update them ALL to reflect the new design. While finding issues in the code is important and useful, it's even more important that the documentation is accurate (and/or you don't have any disagreements with what it says).